### ISSUE TITLE
`unicorn/consistent-destructuring` suggestion should avoid (or warn about) hoisting a var above an `await`

### ISSUE THREAD

**Issue by broofa** [sindresorhus/eslint-plugin-unicorn#1936]
If you open this file in VSCode ...
```js
/* global somePromise */
async function foo() {
  const { baz } = this;

  await somePromise;

  console.log(baz, this.bar);
}

await foo();
```

You'll see a red-squiggly on the `this.bar` text.  Activating the "Replace `this.bar` with destructured `bar`" quick fix will hoist `bar` into the destructuring statement, giving you this:

```js
/* global somePromise */
async function foo() {
  const {baz, bar} = this;

  await somePromise;

  console.log(baz, bar);
}

await foo();
```

The problem with modified code is that bar's value is not guaranteed to be the same pre-await as it is post-await.  (E.g. if `somePromise` is responsible for initializing `this.bar`, then `bar` will hold the uninitialized value.)

It's good that this doesn't happen automatically on `--fix`, but this suggestion should either be disabled  in this case, or include some sort of warning to the user.

**Comment by fregante** [comment]
I think that suggestions, as opposed to autofixes, are for changes that aren't safe, so they should be treated as such by the user. 

I'd hazard to say that side effects are a bad pattern and "somePromise" should not change the value of "this.prop" and not return it.
